### PR TITLE
Akm/romloader2

### PR DESCRIPTION
**Pull request by akm303**
still need to write unit tests, so not pushing through yet, but havent written these yet

### CODE REVIEW

File: src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
+import group1.mips_simulator.Utility;
+import group1.mips_simulator.components.ROM;
+import group1.mips_simulator.components.instructionParts.Instruction;
+import group1.mips_simulator.components.memParts.Memory;
+
+import group1.mips_simulator.components.instructionParts.Instruction;
+import java.util.Vector;
+
 public class RomLoader {
+    Memory memory;
```
Comment (toon-leader-bacon) [inline src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java:12]: It's a little weird to me that a `RomLoader` has-a `Memory`. I'm assuming this is a reference to the same memory that the `Computer` has?

File: src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+    public RomLoader(Memory mem){
+        memory = mem;
     }
-    public RomLoader(String directory){
-        this.ROMLocation = directory;
+
+    public void load(ROM rom){
+        Vector<Instruction> romData = rom.getInstructions();
+        for(int i = 0; i < romData.size(); i++){
+            writeToMemory((short) i, romData.get(i));
```
Comment (toon-leader-bacon) [inline src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java:24]: Can we start this at an offset? Like at memory location 1024 or something? Mem location 0 is reserved.

File: src/main/java/group1/mips_simulator/components/instructionParts/Instruction.java [hunk 3]
```diff
@@ -51,6 +52,24 @@ public String toString_Binary() {
...
 
+
+    public short toShort(){
+        // generates an int from string of self,
+        // returns a short casting of this value
+        int rvalue = 0;
+        char[] instructions = toString_Binary().toCharArray();
+        for(int i = 0; i < instructions.length; i++){
+            if(instructions[instructions.length - i - 1] == '1')
+                rvalue |= 1<<i;
```
Comment (toon-leader-bacon) [inline src/main/java/group1/mips_simulator/components/instructionParts/Instruction.java:63]: Nice. Bit manipulation usually scares me. But this seems ok. Especially if you've got some unit tests here.

File: src/test/java/group1/mips_simulator/UtilityTest.java [hunk 4]
```diff
@@ -44,4 +46,11 @@ void shortToBinaryString_Pretty() {
         assertEquals(Utility.shortToBinaryString_Pretty((short) 15, (short) 16), "0000_0000_0000_1111");
     }
 
+    @Test
+    void instructionToShort() {
+//        Field f1 = new Field(0);
+//        Instruction i1 = new Instruction();
+//        assertEquals(Utility.instructionToShort( ));
```
Comment (toon-leader-bacon) [inline src/test/java/group1/mips_simulator/UtilityTest.java:53]: Why are these commented out?

Comment (toon-leader-bacon) [reply src/test/java/group1/mips_simulator/UtilityTest.java:53]: Oh I see, you just moved them to `InstructionTest.java`. Ok! Looks good

**Review by toon-leader-bacon — APPROVED** [review]
Looks good to me. Approved

File: src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java (hunk 1 above)
Comment (akm303) [reply src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java:12]: yea, it is, though i agree its pretty weird for the romloader to be dedicated to a memory, so ill change that

File: src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java (hunk 2 above)
Comment (akm303) [reply src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java:24]: yep, on it

File: src/main/java/group1/mips_simulator/components/instructionParts/Instruction.java (hunk 3 above)
Comment (akm303) [reply src/main/java/group1/mips_simulator/components/instructionParts/Instruction.java:63]: 🦾

File: src/test/java/group1/mips_simulator/UtilityTest.java (hunk 4 above)
Comment (akm303) [reply src/test/java/group1/mips_simulator/UtilityTest.java:53]: ah yep, ill delete those from the test file

**Pull request merged**